### PR TITLE
Add dysymtab parsing

### PR DESCRIPTION
**Pull request by jrmuizel**
This allows us to have symbols for the imports

### CODE REVIEW

File: src/nc/input/mach-o/MachOParser.cpp [hunk 1]
```diff
@@ -290,9 +326,58 @@ class MachOParserImpl {
...
+                    throw ParseError(tr("Could not read symbol number %1.").arg(i));
+                }
+                byteOrder_.convertFrom(symbolIndex);
+
+                if (! (symbolIndex & INDIRECT_SYMBOL_LOCAL || symbolIndex & INDIRECT_SYMBOL_ABS)) {
+                    if (symbolIndex >= symbols_.size()) {
+                        throw ParseError(tr("symbol index too large %1.").arg(symbolIndex));
+                    }
+                    const core::image::Symbol * sym = symbols_[symbolIndex];
+                    image_->addSymbol(std::make_unique<core::image::Symbol>(sym->type(), sym->name(), indirectSection.addr + i, indirectSection.section));
```
Comment (yegord) [inline src/nc/input/mach-o/MachOParser.cpp]: Are you sure we do not need to add a relocation here, instead of a symbol?

Comment (jrmuizel) [inline src/nc/input/mach-o/MachOParser.cpp]: Yeah, I think it should be a relocation.

Comment (yegord) [inline src/nc/input/mach-o/MachOParser.cpp]: Could you update the pull request and add a test for this?

Comment (jrmuizel) [inline src/nc/input/mach-o/MachOParser.cpp]: Looking at this more closely, it seems like the answer is more nuanced:
dysymtab defines symbols for different sections. The stubs section probably needs actual symbols defined for it, where as the data section should get relocations.

Comment (jrmuizel) [inline src/nc/input/mach-o/MachOParser.cpp]: What form should the test case be in?

Comment (yegord) [inline src/nc/input/mach-o/MachOParser.cpp]: Two files: an input (a Mach-O executable with relocations) and a correct output for it.
See https://github.com/yegord/snowman/blob/master/doc/tests.asciidoc

**Comment by yegord** [comment]
Merged. Thanks!

**Pull request closed without merging**